### DISCUSSION TITLE
Gui Question

### DISCUSSION THREAD

**Discussion by molove** [terjeio/ioSender#168] [Q&A] [answered]
I'm just finishing buiding my openbuilds based cnc using Phil's teensy breakout board.
I'm trying to figure out IOSender. Could somebody tell me what the red squares over axes under "Signal" signify?

![signals](https://user-images.githubusercontent.com/20013346/144680538-64250e74-f919-4633-b8dc-0586991cd6af.jpg)

and why isn't there one over the Y axis? I'm just muddling my way through this so any info would be much appreciated.

Thanks

Piers

**Comment by terjeio** [comment, marked as answer]
It is the status of the limit (and control) input pins (virtual LEDs). With default settings X and Z inputs are not connected to ground, Y is (via  a closed limit switch). You can invert the input readings here:

![image](https://user-images.githubusercontent.com/20260062/144696797-cd543327-2064-4112-9870-5c6de50e8926.png)

For safety reasons limit switches should be "normally closed" (NC) and connected from the input to ground. If wired as "normally open" or not connected you have to invert the pin inputs. I normal operation none of the red squares should show up.

**Comment by molove** [comment]
Thanks, I should have realised it was the limit switches. I've only done a bit of preliminary testing of my control board so far and haven't connected any limit switches yet. 

I've now inverted the limit pins as I'll be using NO switches and sensors, X and Z are working as expected.
I'm using Phil's grblHal build of 3 axes ganged Y and auto-square, so I'm not sure if I should be asking here or on his discussions page,
I can't figure out how the virtual LED for Y works, with the pin inverted the red virtual led is on, but both Y and A are open, closing either or both of them also has no effect on the virtual led.

**Reply by terjeio** [reply]
Does it work without Y ganged? Or with the Y input not inverted and both Y and A connected to ground?

**Comment by molove** [comment]
Sorry, Terjeio, I'm an idiot, I've just realised what is going on. 

On the ganged Y axis I shall be using inductive sensors which because they work at 12v I put a 6.8k pull down resistor between signal and ground, but because I haven't connected the sensors yet the signal pins on the Y and A axes weren't getting 5v but being pulled down
.
 I'm still getting my head around working with Arduino/microprocessors. Hopefully tomorrow I'll have connected the sensors and will be able to test then.

Thanks for taking the time to help me
